devices/src/legacy/i8042.rs

```diff
@@ -6,47 +6,256 @@
 // found in the THIRD-PARTY file.
 
 use logger::{Metric, METRICS};
+use std::fmt;
 use std::{io, result};
 use sys_util::EventFd;
 
 use BusDevice;
 
-const RESET_CMD: u8 = 0xfe;
+#[derive(Debug)]
+pub enum Error {
+    CloneCpuResetEvt(io::Error),
+    KbdInterruptDisabled,
+    KbdInterruptFailure(io::Error),
+}
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::KbdInterruptDisabled => {
+                write!(f, "Keyboard interrupt disabled by guest driver.",)
+            }
+            Error::KbdInterruptFailure(io_err) => {
+                write!(f, "Could not trigger keyboard interrupt: {:?}.", io_err)
+            }
+            Error::CloneCpuResetEvt(io_err) => {
+                write!(f, "Could not clone CPU reset eventfd: {:?}.", io_err)
+            }
+        }
+    }
+}
+
+type Result<T> = result::Result<T, Error>;
+
+/// Offset of the status port (port 0x64)
+const OFS_STATUS: u64 = 4;
+
+/// Offset of the data port (port 0x60)
+const OFS_DATA: u64 = 0;
+
+/// i8042 commands
+/// These values are written by the guest driver to port 0x64.
+const CMD_READ_CTR: u8 = 0x20; // Read control register
+const CMD_WRITE_CTR: u8 = 0x60; // Write control register
+const CMD_READ_OUTP: u8 = 0xD0; // Read output port
+const CMD_WRITE_OUTP: u8 = 0xD1; // Write output port
+const CMD_RESET_CPU: u8 = 0xFE; // Reset CPU
+
+/// i8042 status register bits
+const SB_OUT_DATA_AVAIL: u8 = 1 << 0; // Data available at port 0x60
+const SB_I8042_CMD_DATA: u8 = 1 << 3; // i8042 expecting command parameter at port 0x60
+const SB_KBD_ENABLED: u8 = 1 << 4; // 1 = kbd enabled, 0 = kbd locked
+
+/// i8042 control register bits
+const CB_KBD_INT: u8 = 1 << 0; // kbd interrupt enabled
```

Comment:
The kernel names are rather confusing and I found them pretty difficult to follow. The whole i8042 kernel code is huge, since it has to deal with all kinds of buggy old hardware, while I'm only implementing a very small subset of the i8042 functionality here. Hence, I wouldn't recommend reading the two side-by-side. It would probably be much easier to follow an i8042 guide, such as [this one](http://stanislavs.org/helppc/8042.html), or [this one](https://wiki.osdev.org/%228042%22_PS/2_Controller).

Reply:
I stand by my point, which boils down to: naming things the same as in the kernel (however convoluted those names may be) simplifies (new) developers' lives by making it easier to match Firecracker code with the kernel functionalities.
I wouldn't recommend reading the i8042 kernel code in order to understand Firecracker code, and I agree that putting them side by side doesn't paint a pretty picture, but I would find it useful to make my way quickly from the slim Firecracker code to the kernel (and I have memories of struggling with virtio in my early Firecracker days because it wasn't clear from anywhere how things interacted - and constants had different names :disappointed: )